File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 19/VB/DataBindinginSilverlightVB/DataBindinginSilverlightVB.Web/MyService.svc.vb
'NOTE: If you change the class name "MyService" here, you must also update the reference to "MyService" in Web.config and in the associated .svc file.
Public Class MyService
    Implements IMyService

    'Public Sub DoWork() Implements IMyService.DoWork
    'End Sub

    'Public Sub GetByLastName() Implements IMyService.GetByLastName

    'End Sub

	Public Function GetByLastName(ByVal lastname As String) As System.Collections.Generic.List(Of Employee) Implements IMyService.GetByLastName
        Dim db As DataBindinginSilverlightVB.Web.MyClassDataContext = New MyClassDataContext()
        Dim data = From p In db.Employees Where p.LastName.StartsWith(lastname) Select p
        Return data.ToList
    End Function
End Class

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 19/CS/DataBindinginSilverlightCS/DataBindinginSilverlightCS.Web/MyService.svc.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.Text;

namespace DataBindinginSilverlightCS.Web
{
    // NOTE: You can use the "Rename" command on the "Refactor" menu to change the class name "MyService" in code, svc and config file together.
    public class MyService : IMyService
    {
        public void DoWork()
        {
        }

        #region IMyService Members
        public List<Employee> GetByLastName(string lastname)
        {
           // throw new NotImplementedException();
            DataBindinginSilverlightCS.Web.MyClassDataContext db = new MyClassDataContext();
            var data = from p in db.Employees
                       where p.LastName.StartsWith(lastname)
                       select p;
            return data.ToList();
        }
        #endregion 
    }
}
